contracts/OrderMixin.sol

```diff
@@ -245,13 +249,23 @@ abstract contract OrderMixin is IOrderMixin, EIP712, PredicateHelper {
         }
 
         // Maker => Taker
-        if (!_callTransferFrom(
-            order.makerAsset.get(),
-            order.maker.get(),
-            target,
-            actualMakingAmount,
-            order.makerAssetData()
-        )) revert TransferFromMakerToTakerFailed();
+        if ((skipPermitUsePermit2ThresholdAmount & _USE_PERMIT2_FLAG) != 0) {
```

Comment:
skipPermitUsePermit2ThresholdAmount should impact taker => maker flow.

maker => taker flow should be controlled by the maker and can be set in the top bit of makerAsset.